### PR TITLE
Fix retries=0 bug in PortUtils.ConnectionCheckSSH

### PR DESCRIPTION
**Pull request by pjdarton**
* Issue #751 - Math.min should've been Math.max
* Removed unused imports from unit test
* Fixed failure-case unit test for ConnectionCheckSSH

### CODE REVIEW

File: src/main/java/com/nirima/jenkins/plugins/docker/utils/PortUtils.java [hunk 1]
```diff
@@ -157,7 +157,7 @@ public ConnectionCheckSSH withSSHTimeout(int time, TimeUnit units) {
         public boolean execute() throws InterruptedException {
             checkState(parent.execute(), "Port %s is not opened to connect to", parent.port);
 
-            final int retries = Math.min(0, parent.retries);
+            final int retries = Math.max(0, parent.retries);
```
Comment (pjdarton) [inline src/main/java/com/nirima/jenkins/plugins/docker/utils/PortUtils.java]: This is the bugfix.

File: src/main/java/com/nirima/jenkins/plugins/docker/utils/PortUtils.java [hunk 2]
```diff
@@ -172,16 +172,16 @@ public boolean execute() throws InterruptedException {
         }
 
         private boolean executeOnce(final int thisTryNumber, final int totalTriesIntended) {
-            final Connection connection = new Connection(parent.host, parent.port);
+            final Connection sshConnection = new Connection(parent.host, parent.port);
```
Comment (pjdarton) [inline src/main/java/com/nirima/jenkins/plugins/docker/utils/PortUtils.java]: I renamed the variable to make it clear that this was a SSH connection not merely a TCP connection that was being made.

File: src/test/java/com/nirima/jenkins/plugins/docker/utils/PortUtilsTest.java [hunk 3]
```diff
@@ -7,18 +7,15 @@
 
 import java.io.IOException;
 import java.net.ServerSocket;
-import java.util.Date;
```
Comment (pjdarton) [inline src/test/java/com/nirima/jenkins/plugins/docker/utils/PortUtilsTest.java]: Removed some unused imports

File: src/test/java/com/nirima/jenkins/plugins/docker/utils/PortUtilsTest.java [hunk 4]
```diff
@@ -7,18 +7,15 @@
...
 import java.util.concurrent.Executors;
 import java.util.concurrent.TimeUnit;
 
 import static java.lang.System.currentTimeMillis;
 import static java.util.concurrent.TimeUnit.MILLISECONDS;
 import static java.util.concurrent.TimeUnit.SECONDS;
 import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.Matchers.both;
+import static org.hamcrest.Matchers.equalTo;
```
Comment (pjdarton) [inline src/test/java/com/nirima/jenkins/plugins/docker/utils/PortUtilsTest.java]: Replaced use of `is` with `equalTo` as "is" is deprecated.

File: src/test/java/com/nirima/jenkins/plugins/docker/utils/PortUtilsTest.java [hunk 5]
```diff
@@ -79,19 +76,24 @@ public void shouldThrowIllegalStateExOnNotAvailPort() throws Exception {
     @Test
     public void shouldWaitIfPortAvailableButNotSshUntilTimeout() throws Exception {
```
Comment (pjdarton) [inline src/test/java/com/nirima/jenkins/plugins/docker/utils/PortUtilsTest.java]: This was the buggy unit test that allowed the bug to go undetected.

**Comment by pjdarton** [comment]
@php50350 @biozshock This is the proposed fix for issue #751.
Please download the `.hpi` file from the "Last Successful Artifacts" section in https://ci.jenkins.io/job/Plugins/job/docker-plugin/job/PR-752/, install it on your Jenkins server(s) (via manage jenkins -> manage plugins -> advanced), test that it fixes the bug, and let me know the results (either here or in #751).

Once the fix is confirmed I'll merge the changes.

**Comment by biozshock** [comment]
@pjdarton the URL does not work for me:
$ telnet ci.jenkins.io 443
Trying 104.208.238.39...
telnet: Unable to connect to remote host: Connection refused

same with http

**Comment by pjdarton** [comment]
@biozshock That's weird.  Maybe there's some internet firewall preventing you accessing that server? That server should be open to everyone on the internet...

Anyway, I've attached (a zip containing) the .hpi file here, [docker-plugin-1.1.9-rc895.e262f2331080.hpi.zip](https://github.com/jenkinsci/docker-plugin/files/3676335/docker-plugin-1.1.9-rc895.e262f2331080.hpi.zip).  Hopefully you'll be able to download it from here.
(I had to wrap it in a zip file because github says it doesn't allow uploads of .hpi files, even though these are really zip files by another name)

**Comment by biozshock** [comment]
Seems to be working correctly here. Thanks for quick fix @pjdarton

**Comment by dschnabel** [comment]
We ran into the same issue (#751) with 1.1.8 and the 1.1.9 RC as provided above fixed it for us. Thank you.

**Comment by pjdarton** [comment]
Code changes went into released of docker-plugin version 1.1.9

**Pull request merged**